### PR TITLE
Allow checking in with show, season and episode

### PR DESCRIPTION
**Pull request by samicemalone**
Allow checking in with show, season and episode without needing EpisodeIds.

Added Show to EpisodeCheckin to allow searching by name, ids etc.. as per the [docs example](http://docs.trakt.apiary.io/#reference/checkin/checkin/check-into-an-item)

Added season number to SyncEpisode

### CODE REVIEW

**Comment by UweTrottmann** [comment]
Looks good. However, could you add [a test](https://github.com/UweTrottmann/trakt-java/blob/master/src/test/java/com/uwetrottmann/trakt/v2/services/CheckinTest.java) for this?

P.S. Sorry for the late response.

**Comment by samicemalone** [comment]
Hi,

Thanks for the feedback. I have added the test case for this. It was passing when using my own client id and access token.

Thanks

File: src/test/java/com/uwetrottmann/trakt/v2/services/CheckinTest.java [hunk 1]
```diff
@@ -2,15 +2,7 @@
...
-import com.uwetrottmann.trakt.v2.entities.EpisodeCheckin;
-import com.uwetrottmann.trakt.v2.entities.EpisodeCheckinResponse;
-import com.uwetrottmann.trakt.v2.entities.EpisodeIds;
-import com.uwetrottmann.trakt.v2.entities.MovieCheckin;
-import com.uwetrottmann.trakt.v2.entities.MovieCheckinResponse;
-import com.uwetrottmann.trakt.v2.entities.MovieIds;
-import com.uwetrottmann.trakt.v2.entities.ShareSettings;
-import com.uwetrottmann.trakt.v2.entities.SyncEpisode;
-import com.uwetrottmann.trakt.v2.entities.SyncMovie;
+import com.uwetrottmann.trakt.v2.entities.*;
```
Comment (UweTrottmann) [inline src/test/java/com/uwetrottmann/trakt/v2/services/CheckinTest.java]: One more thing: no star imports please :) In most cases your IDE is configured to do this automatically (it's good practice to turn it off).

File: src/test/java/com/uwetrottmann/trakt/v2/services/CheckinTest.java [hunk 2]
```diff
@@ -51,6 +43,30 @@ public void test_checkin_episode() throws OAuthUnauthorizedException {
...
+        try {
+            response = getTrakt().checkin().checkin(checkin);
+        } catch (CheckinInProgressException e) {
+            fail("Check-in still in progress, may be left over from failed test");
+        }
+
+        // delete check-in first
+        test_checkin_delete();
+
+        assertThat(response).isNotNull();
```
Comment (UweTrottmann) [inline src/test/java/com/uwetrottmann/trakt/v2/services/CheckinTest.java]: Also would be nice if you extract the assertions into a common helper method.

**Comment by UweTrottmann** [comment]
Added two more remarks, minor stuff. If you don't have the time, let me know and I do it for you!

**Comment by samicemalone** [comment]
Sorry about the delay with this! I've fixed my IDE now so the star imports shouldn't happen again!

Cheers

**Comment by UweTrottmann** [comment]
Awesome, thanks! Merging.

**Pull request merged**